### PR TITLE
Sign payload for sovereign builder

### LINKED ISSUE #170: Sovereign Sequencer v1: ensure only sovereign sequencer can sequence L2 blocks
For the marketplace version this is handled by the marketplace. Only the designated builder can build blocks for the namespace and the rollup operator can operate the builder.

For the non-marketplace version there needs to be a way for external parties to verify that a transaction in the namespace is part of the canonical L2 chain because anyone can send transactions to the namespace.

One way to achieve this is by signing the payload of the Espresso transaction that the sovereign sequencer sends to the `/submit` endpoint.

I suggest that we work on this last for implementing the v1 changes because we can test everything else without it and there are some decisions to be made as to what key to use and how to distribute and rotate it. So if we're lucky by the time we would have to implement this we already went to v2.

If we do need to implement this I think it would make sense to re-use the wallet of the `DataPoster` for signing because it's also the entity that signs the batchers L1 transactions. We can potentially use the `datasigner` to sign our transaction bytes which is how nitro seems to sign messages it broadcasts between services.

https://github.com/EspressoSystems/nitro-espresso-integration/blob/ee9c8674a969758e57a187975b88ba56c54b8b69/util/signature/datasigner.go#L12

### PR DESCRIPTION
**Pull request by Sneh1999**
Closes #170

https://github.com/EspressoSystems/nitro-espresso-integration/issues/170
### This PR:

Signs payload before sending it using `/submit` endpoint

### CODE REVIEW

File: arbnode/transaction_streamer.go
Comment (ImJeremyHe) [inline arbnode/transaction_streamer.go:1468]: I think panic directly may be better

File: arbnode/transaction_streamer.go
Comment (ImJeremyHe) [inline arbnode/transaction_streamer.go:1483]: I think it is a bit weird here. I think we should also sign the namespace. And maybe hotshot transaction should have a field for signature.

**Review by ImJeremyHe — APPROVED** [review]
Just some comments.

File: arbnode/transaction_streamer.go
Comment (sveitser) [reply arbnode/transaction_streamer.go:1483]: Ideally we would sign the whole transaction including the namespace. However, concretely I can't think of "replaying" the tx in another namespace causing any problems, as long as we don't sign the namespace unless someone is using the same key for two namespaces. So we should leave a not about this restriction. Let's keep in mind that we will likely remove this signing feature soon anyway and not sink more time into it than necessary.

It does feel like we should have a test somewhere that does the signature recovery so that we can be somewhat sure it would work for 3rd parties that are watching the Espresso transaction stream and want to verify signatures.

Additionally I think we should add some comments to the code as to "why" we are signing the messages and that internally the signature doesn't need to verified because the batch poster receives the transactions from the sequencer. Unless that's not current, then please tell me.

cc @Sneh1999 @ImJeremyHe @zacshowa

File: arbnode/transaction_streamer.go
Comment (sveitser) [reply arbnode/transaction_streamer.go:1468]: Agreed. If something is wrong that we can't handle in the code it's best to exit as early as possible with a clear error message explaining the users why we're crashing.

File: arbnode/transaction_streamer.go
Comment (Sneh1999) [reply arbnode/transaction_streamer.go:1485]: @sveitser @ImJeremyHe  added test and comments! Feel free to take a look when you have sometime

File: system_tests/espresso_transaction_payload_signature_test.go
Comment (ImJeremyHe) [inline system_tests/espresso_transaction_payload_signature_test.go:13]: I would like to make this test simpler, but so far I don't have any clue

**Review by ImJeremyHe — APPROVED** [review]

File: system_tests/espresso_transaction_payload_signature_test.go
Comment (sveitser) [reply system_tests/espresso_transaction_payload_signature_test.go:13]: @ImJeremyHe @Sneh1999 based on the discussion yesterday. We will likely implement an Espresso pre-confirmation node that finalizes L2 blocks when Espresso finalized them. If we are still on sovereign sequencer v1 at that time let's implement this signing and test it. If things go well and we have the v2 we don't need to bother with the signing. So I would suggest we keep this PR open but stop working on it and don't merge it right now then finish it later if necessary, or hopefully just close it.

**Review by zacshowa — APPROVED** [review]
The code LGTM for mergeing once CI is satisfied. I think we should also revisit the items that Jeremy and Mathis mentioned in August.

File: arbnode/transaction_streamer.go
Comment (zacshowa) [inline arbnode/transaction_streamer.go:1485]: Perhaps we can refactor the logic in the transaction streamer into a separate method, or an external function. This method could take a list of txns, a dataSigner function, and the namespace. It would  build and return the transaction to be passed to the `espressoClient.SubmitTransaction` method. 

This way we could write a unit test over the function rather than having to run a network to observe the results. We could also do this for just the payload itself.

Comment (Sneh1999) [reply arbnode/transaction_streamer.go:1485]: IMO, there is still sense in testing the end to end flow because if we create the test centric to this function we will not be able to test the rest of the parts of the PR like changes in `parsing` logic in arbos/parse_l2.go

**Pull request merged**
